src/ContractGenerator/ContractGenerator.cs

```diff
@@ -20,20 +21,236 @@ public class FlagConstants
+    ///     ServicesFilename generates Services FileName based on the FileDescriptor
+    ///     Its based on the C++ original
+    ///     https://github.com/AElfProject/contract-plugin/blob/de625fcb79f83603e29d201c8488f101b40f573c/src/contract_csharp_generator_helpers.h#L27
     /// </summary>
-    private static string GetServicesFilename(FileDescriptor fileDescriptor)
+    private static string ServicesFilename(IDescriptor file)
     {
-        throw new NotImplementedException();
+        return FileNameInUpperCamel(file, false) + ".c.cs";
+    }
+
+    private static string LowerUnderscoreToUpperCamel(string str)
+    {
+        var tokens = str.Split('_');
+        var result = new StringBuilder();
+
+        foreach (var token in tokens) result.Append(CapitalizeFirstLetter(token));
+
+        return result.ToString();
+    }
+
+    private static string FileNameInUpperCamel(IDescriptor file, bool includePackagePath)
+    {
+        var tokens = StripProto(file.Name).Split('/');
+        var result = new StringBuilder();
+
+        if (includePackagePath)
+            for (var i = 0; i < tokens.Length - 1; i++)
+                result.Append(tokens[i] + "/");
+
+        result.Append(
+            LowerUnderscoreToUpperCamel(tokens[^1])); // Using the "end index" operator to get the last element
+
+        return result.ToString();
+    }
+
+    private static string CapitalizeFirstLetter(string str)
+    {
+        if (string.IsNullOrEmpty(str)) return str;
+
+        var chars = str.ToCharArray();
+        chars[0] = char.ToUpper(chars[0]);
+        return new string(chars);
+    }
+
+    private static string StripProto(string fileName)
+    {
+        return fileName.EndsWith(".proto") ? fileName[..^".proto".Length] : fileName;
     }
 
     /// <summary>
     ///     Generates a set of C# files from the input stream containing the proto source. This is the primary entry-point into
     ///     the ContractPlugin.
     /// </summary>
-    public CodeGeneratorResponse Generate(Stream stdin)
+    public static CodeGeneratorResponse Generate(Stream stdin)
+    {
+        var response = new CodeGeneratorResponse();
+        CodeGeneratorRequest request;
+        IReadOnlyList<FileDescriptor> fileDescriptors;
+
+        using (stdin)
+        using (var memoryStream = new MemoryStream())
+        {
+            stdin.CopyTo(memoryStream);
+            memoryStream.Seek(0, SeekOrigin.Begin);
+            request = Deserialize<CodeGeneratorRequest>(memoryStream);
+            // need to rewind the stream before we can read again
+            memoryStream.Seek(0, SeekOrigin.Begin);
+            fileDescriptors = FileDescriptorSetLoader.Load(memoryStream);
+        }
+
+        var flags = FlagConstants.GenerateContractWithEvent;
+        var options = new List<Tuple<string, string>>();
+        ProtoUtils.ParseGeneratorParameter(request.Parameter, options);
```

Comment:
As discussed with steven

Move all this param-option mapping logic below + the stdin-parsing stuff into the ContractPlugin Program.cs. then change this func signature to just `CodeGeneratorResponse Generate(FileDescriptor file, byte flags)`. The request object can be removed.

For the option params we can refer to https://github.com/AElfProject/AElf/blob/bd91fc5e23f03f5179cee9fc6cfd5a0ee5b802d2/scripts/generate_event_only.sh#L13 for examples of how to pass down these options as params into protoc. and modify the testcase python script to include them too